### PR TITLE
Improve `ZydisRegisterGetLargestEnclosing`

### LINKED ISSUE #443: Wrong output of ZydisRegisterGetLargestEnclosing
Function ZydisRegisterGetLargestEnclosing can't return the correct output for a variety of registers. Here is the proof code:

    #include <stdio.h>
    #include <Zydis/Zydis.h>
    
    
    int main()
    {
        // Same with ZYDIS_REGISTER_FLAGS, ZYDIS_REGISTER_IP, ZYDIS_REGISTER_EIP (maybe there are more)
        ZydisRegister result = ZydisRegisterGetLargestEnclosing(ZYDIS_MACHINE_MODE_LONG_64, ZYDIS_REGISTER_EFLAGS);
    
        if (result == ZYDIS_REGISTER_NONE)
            puts("ZydisRegisterGetLargestEnclosing returned: ZYDIS_REGISTER_NONE");
    
        else if (result == ZYDIS_REGISTER_RFLAGS)
            puts("ZydisRegisterGetLargestEnclosing returned: ZYDIS_REGISTER_RFLAGS");
    
        else
            printf("ZydisRegisterGetLargestEnclosing returned unxepected return value: %s\n",
                ZydisRegisterGetString(result));
    
    
        return 0;
    }

Same for Rust bindings: https://github.com/zyantific/zydis-rs/issues/32

### PR DESCRIPTION
**Pull request by flobernd**
Closes #443

### CODE REVIEW

**Review by athre0z — APPROVED** [review]
Thanks -- looks good to me! :)

File: src/Register.c [hunk 1]
```diff
@@ -156,6 +156,67 @@ ZydisRegister ZydisRegisterGetLargestEnclosing(ZydisMachineMode mode, ZydisRegis
...
+        /* TMM     */ { ZYDIS_REGISTER_NONE , ZYDIS_REGISTER_NONE  , ZYDIS_REGISTER_NONE   },
+        /* FLAGS   */ { ZYDIS_REGISTER_FLAGS, ZYDIS_REGISTER_EFLAGS, ZYDIS_REGISTER_RFLAGS },
+        /* IP      */ { ZYDIS_REGISTER_IP   , ZYDIS_REGISTER_EIP   , ZYDIS_REGISTER_RIP    },
+        /* SEGMENT */ { ZYDIS_REGISTER_NONE , ZYDIS_REGISTER_NONE  , ZYDIS_REGISTER_NONE   },
+        /* TABLE   */ { ZYDIS_REGISTER_NONE , ZYDIS_REGISTER_NONE  , ZYDIS_REGISTER_NONE   },
+        /* TEST    */ { ZYDIS_REGISTER_NONE , ZYDIS_REGISTER_NONE  , ZYDIS_REGISTER_NONE   },
+        /* CONTROL */ { ZYDIS_REGISTER_NONE , ZYDIS_REGISTER_NONE  , ZYDIS_REGISTER_NONE   },
+        /* DEBUG   */ { ZYDIS_REGISTER_NONE , ZYDIS_REGISTER_NONE  , ZYDIS_REGISTER_NONE   },
+        /* MASK    */ { ZYDIS_REGISTER_NONE , ZYDIS_REGISTER_NONE  , ZYDIS_REGISTER_NONE   },
+        /* BOUND   */ { ZYDIS_REGISTER_NONE , ZYDIS_REGISTER_NONE  , ZYDIS_REGISTER_NONE   }
```
[the remark this answers is not in the document]
Comment (flobernd) [reply src/Register.c:194]: Sounds good! Seems like my logic is flawed as well as the compiler complains about unreachable code 😓

File: src/Register.c [hunk 2]
```diff
@@ -156,6 +156,67 @@ ZydisRegister ZydisRegisterGetLargestEnclosing(ZydisMachineMode mode, ZydisRegis
...
+    {
+    case ZYDIS_MACHINE_MODE_LONG_64:
+        mode_bits = 2;
+    case ZYDIS_MACHINE_MODE_LONG_COMPAT_32:
+    case ZYDIS_MACHINE_MODE_LEGACY_32:
+        mode_bits = 1;
+    case ZYDIS_MACHINE_MODE_LONG_COMPAT_16:
+    case ZYDIS_MACHINE_MODE_LEGACY_16:
+    case ZYDIS_MACHINE_MODE_REAL_16:
+        mode_bits = 0;
```
Comment (flobernd) [inline src/Register.c:209]: Ah, the breaks are missing in this switch.

**Comment by ZehMatt** [comment]
Also good opportunity to add tests for this maybe?

**Review by athre0z — APPROVED** [review]

**Pull request merged**